### PR TITLE
Integrating changes into Hyrax 2.x releases for supporting BrowseEverything 1.0.x

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Ports the changes proposed in https://github.com/samvera/hyrax/pull/3645 for the 2.x release series.

### CODE REVIEW

File: hyrax.gemspec [hunk 1]
```diff
@@ -37,8 +37,7 @@ SUMMARY
   spec.add_dependency 'blacklight', '~> 6.14'
   spec.add_dependency 'blacklight-gallery', '~> 0.7'
   spec.add_dependency 'breadcrumbs_on_rails', '~> 3.0'
-  # Pin browse-everything to stable version
-  spec.add_dependency 'browse-everything', '< 0.16'
+  spec.add_dependency 'browse-everything', '~> 1.0'
```
Comment (no-reply) [inline hyrax.gemspec]: I'm not sure it's okay for us to force this upgrade as part of the 2.x release series. Thoughts?

Comment (jrgriffiniii) [reply hyrax.gemspec]: Certainly, I can adjust this PR.

**Comment by jrgriffiniii** [comment]
This is still blocked until https://github.com/samvera/browse-everything/pull/270/files is reviewed and merged upstream.

**Comment by jrgriffiniii** [comment]
This is now ready for review by the @samvera/hyrax-code-reviewers please

**Comment by jrgriffiniii** [comment]
The tests on this should now properly the cover the changes which are introduced with this upgrade.

**Comment by jrgriffiniii** [comment]
Blocked by https://github.com/samvera/hyrax/pull/3843

**Comment by orangewolf** [comment]
@jrgriffiniii looks like we are down the the gemspec version numbers. Does that still need to be loosened? Otherwise this looks unblocked and ready.

**Comment by jrgriffiniii** [comment]
1.1.0 has yet to be released, and https://wiki.duraspace.org/display/samvera/Samvera+BrowseEverything+Interest+Group will ensure that semantic versioning is followed for future releases.  I'll adjust this now.

**Comment by jrgriffiniii** [comment]
Flapping tests are now passing: https://circleci.com/workflow-run/1ee8b65b-3e7c-4c0c-b1df-1a41a0f11909

**Comment by scherztc** [comment]
@jrgriffiniii 👍  I tested this branch SUCCESSFULLY with the .internal_test app using our credentials for Google Drive, DropBox, Box, and File System.

**Comment by jrgriffiniii** [comment]
Please see https://github.com/samvera/hyrax/pull/3645#discussion_r299968957 for some needed improvements, but this was not considered blocking for that PR.

**Comment by jrgriffiniii** [comment]
I'm going to please request input before merging from @no-reply, @cjcolvar, or others on @samvera/hyrax-code-reviewers given the question in https://github.com/samvera/hyrax/pull/3652#discussion_r266962426.  As much as I don't want to block this any further, but I just want to be certain that there are no outstanding concerns from active contributors who have reviewed this in the past.

File: hyrax.gemspec [hunk 2]
```diff
@@ -37,8 +37,7 @@ SUMMARY
   spec.add_dependency 'blacklight', '~> 6.14'
   spec.add_dependency 'blacklight-gallery', '~> 0.7'
   spec.add_dependency 'breadcrumbs_on_rails', '~> 3.0'
-  # Pin browse-everything to stable version
-  spec.add_dependency 'browse-everything', '< 0.16'
+  spec.add_dependency 'browse-everything', '< 2.0'
```
Comment (cjcolvar) [inline hyrax.gemspec]: I think this might need to be '>= 0.16', '< 2.0' since it looks like `Retriever#can_retrieve?` didn't appear until 0.16.0.

**Pull request merged**